crates/sui-config/src/node.rs

```diff
@@ -33,19 +35,38 @@ pub const DEFAULT_GRPC_CONCURRENCY_LIMIT: usize = 20000000000;
 #[serde(rename_all = "kebab-case")]
 pub struct NodeConfig {
     /// The keypair that is used to deal with consensus transactions
-    #[serde(default = "default_key_pair")]
-    #[serde_as(as = "Arc<KeyPairBase64>")]
-    pub protocol_key_pair: Arc<AuthorityKeyPair>,
+    #[serde(default = "default_key_pair", skip_serializing_if = "Option::is_none")]
+    #[serde_as(as = "Option<Arc<KeyPairBase64>>")]
+    pub protocol_key_pair: Option<Arc<AuthorityKeyPair>>,
     /// The keypair that is used by the narwhal worker.
-    #[serde(default = "default_worker_key_pair")]
-    #[serde_as(as = "Arc<KeyPairBase64>")]
-    pub worker_key_pair: Arc<NetworkKeyPair>,
+    #[serde(
+        default = "default_worker_key_pair",
+        skip_serializing_if = "Option::is_none"
+    )]
+    #[serde_as(as = "Option<Arc<KeyPairBase64>>")]
+    pub worker_key_pair: Option<Arc<NetworkKeyPair>>,
     /// The keypair that the authority uses to receive payments
-    #[serde(default = "default_sui_key_pair")]
-    pub account_key_pair: Arc<SuiKeyPair>,
-    #[serde(default = "default_worker_key_pair")]
-    #[serde_as(as = "Arc<KeyPairBase64>")]
-    pub network_key_pair: Arc<NetworkKeyPair>,
+    #[serde(
+        default = "default_sui_key_pair",
+        skip_serializing_if = "Option::is_none"
+    )]
+    pub account_key_pair: Option<Arc<SuiKeyPair>>,
+    #[serde(
+        default = "default_worker_key_pair",
+        skip_serializing_if = "Option::is_none"
+    )]
+    #[serde_as(as = "Option<Arc<KeyPairBase64>>")]
+    pub network_key_pair: Option<Arc<NetworkKeyPair>>,
+
+    /// File path to read the protocol_key_pair from.
+    pub protocol_key_pair_path: PathBuf,
+    /// File path to read the worker_key_pair from.
+    pub worker_key_pair_path: PathBuf,
+    /// File path to read the account_key_pair from.
+    pub account_key_pair_path: PathBuf,
+    /// File path to read the network_key_pair from.
+    pub network_key_pair_path: PathBuf,
```

Comment:
Instead of having two fields, one for a file path and one for an embedded key, could we improve the UX of this to instead be handled by a single config like we've done for genesis? https://github.com/MystenLabs/sui/blob/main/crates/sui-config/src/node.rs#L300-L306

Reply:
revised